addons/enemybar/actionTracking.lua

```diff
@@ -0,0 +1,299 @@
+			if is_pet then
+				if p.mob.pet_index == mob.index then return true end
+			else
+				if p.mob.id == mob.id then return true end
+			end
+		end
+	end
+	for i=0, party.party3_count-1 do
+		local p = party['a2'..i]
+		if p and p.mob then
+			if is_pet then
+				if p.mob.pet_index == mob.index then return true end
+			else
+				if p.mob.id == id then return true end
+			end
+		end
+	end
+
+    return false
+end
+
+function handle_action_packet(id, data)
+	if 0x028 == id then
+		local ai = windower.packets.parse_action(data)
+
+		track_enmity(ai)
+		track_actions(ai)
+		track_debuffs(ai)
+	elseif 0x029 then
+    	local message_id = data:unpack('H',0x19)
+    	if not message_id then return end
+    	message_id = message_id%32768
+		local param_1 = data:unpack('I',0x0D)
+    	local target_id = data:unpack('I',0x09)
+		if S{204,206}:contains(message_id) then
+			-- wears off message.
+			local target = windower.ffxi.get_mob_by_id(target_id)
+			if target and tracked_debuff[target.id]  then
+				tracked_debuff[target.id][param_1] = nil
+			end
+		end
+	end
+end
+
+function track_enmity(ai)
+	local actor = windower.ffxi.get_mob_by_id(ai.actor_id)
+	if not actor then return end
+
+	for i,t in ipairs(ai.targets) do
+		local target = windower.ffxi.get_mob_by_id(t.id)
+		if target then
+			local pc = nil
+			local mob = nil
+			if actor.is_npc and not actor.charmed and (not target.is_npc or target.charmed) then
+				pc = target
+				mob = actor
+			elseif (target.is_npc and not target.charmed) and (not actor.is_npc or target.charmed)  then
+				mob = target
+				pc = actor
+			end
+
+			if pc and mob and is_party_member_or_pet(pc) then
+				-- we have npc/pc interaction
+				-- if the actor is the npc, then we know there's enmity. Otherwise, if the target of the pc spell isn't tracked, it definitely hates the pc now.
+				if actor == mob or not tracked_enmity[mob.id] then
+					tracked_enmity[mob.id] = {pc=pc, mob=mob, time=os.time()}
+
+					-- if the actor is the npc, we don't care about the other targets. The first target is the one they targeted.
+					if actor == mob then return end
+				end
+			end
+		end
+	end
+end
+
+function track_actions(ai)
+	local actor = windower.ffxi.get_mob_by_id(ai.actor_id)
+	if not actor then return end
+
+	local target = windower.ffxi.get_mob_by_id(ai.targets[1].id)
+	if not target then
+		target = {name='Unknown (id:'..ai.targets[1].id, id=ai.targets[1].id}
+	end
+
+	-- if the category is not casting magic, jas, items or ws, don't bother.
+	if not S{8,4,7,11,3,6,9,5}:contains(ai.category) then return end
+
+	-- if it's a starting packet, the id is in param2
+	local action_id = ai.param
+	if S{8,7,9}:contains(ai.category) then
+		action_id = ai.targets[1].actions[1].param
+	end
+
+	-- find the action
+	local action_map = nil
+	if S{8,4}:contains(ai.category) then
+		action_map = res.spells[action_id]
+	elseif S{5,9}:contains(ai.category) then
+		action_map = res.items[action_id]
+	else
+		if actor.is_npc then
```

Comment:
You can combine this and the previous line into an `elseif`. Same for the check below.